src/seedu/addressbook/AddressBook.java

```diff
@@ -305,12 +308,45 @@ private static void setupDefaultFileForStorage() {
     }
 
     /**
-     * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
-     * TODO: Implement a more rigorous validity checking.
+     * Returns true if the given file path is valid.
+     * A file path is valid if it has a valid parent directory as determined by {@link #hasValidFolder}
+     * and a valid file name as determined by {@link #hasValidFileName}.
      */
     private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+        if (filePath == null) {
+            return false;
+        }
+        try {
+            Path filePathToValidate = Paths.get(filePath);
+            return hasValidParentDirectory(filePathToValidate) && hasValidFileName(filePathToValidate);
+        } catch (InvalidPathException ipe) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if the file path has a parent directory that exists
+     */
+    private static boolean hasValidParentDirectory(Path filePath) {
+        Path parentDirectory = filePath.getParent();
+        if (parentDirectory == null) {
+            return true;
+        }
+        return Files.isDirectory(parentDirectory, LinkOption.NOFOLLOW_LINKS) 
+                && Files.exists(parentDirectory, LinkOption.NOFOLLOW_LINKS);
+    }
+
+    /**
+     * Returns true if file path has a valid file name.
+     * File name is valid if it has an extension and no reserved characters.
+     * Reserved characters are OS-dependent.
+     */
+    private static boolean hasValidFileName(Path filePath) {
+        if (Files.isDirectory(filePath, LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
Hrm, I'm guessing you want something like "If the file path exists, then it must be a regular file".

A file not being a directory does not mean that it is a regular file though. On POSIX systems for example files can also be unix domain sockets or FIFO files.

So, in combination with the lines below, you probably want:
```java
return filePath.getFileName().toString().lastIndexOf('.') > 0 && (!Files.exists(filePath) || Files.isRegularFile(filePath));
```

By the way, this new requirement needs to be stated in the javadoc as well.